### PR TITLE
Remove enzyme and translate any remaining tests to RTL

### PR DESCRIPTION
**Pull request by jeremywiebe**
## Summary:

While working on our WB deps issues, I noticed we still use Enzyme for a few tests.  Two were `xit()` (ignored) in the math-input project so I deleted them as they were testing our legacy keypad components. The rest were easily migrated to our utility `renderQuestion()` function and Testing Library!

Issue: "none"

## Test plan:

`yarn test`

### CODE REVIEW

File: config/test/custom-matchers.ts [hunk 1]
```diff
@@ -1,17 +1,9 @@
 // Ok here we are
 
-type Score =
```
Comment (jeremywiebe) [inline config/test/custom-matchers.ts:3]: This type didn't match our actual Score type.

File: config/test/custom-matchers.ts [hunk 2]
```diff
@@ -32,7 +24,7 @@ expect.extend({
                 message: () => `Problem was not fully answered`,
             };
         }
-        if (score.earned !== 1) {
```
Comment (jeremywiebe) [inline config/test/custom-matchers.ts:35]: And incorrect score is one where the `earned` is not equal to `.total`.  Some questions have more than one "point" available.

File: packages/perseus/src/renderer.tsx [hunk 3]
```diff
@@ -132,7 +132,7 @@ export type Widget = {
         path: FocusPath,
         newValue: string,
         // TODO(jeremy): I think this is actually a callback
-        focus: () => unknown,
+        focus?: () => unknown,
```
Comment (jeremywiebe) [inline packages/perseus/src/renderer.tsx:135]: This callback/focus function is truly optional.  The API tests in this PR illustrate that by not passing it sometimes.

**Review by benchristel — APPROVED** [review]
LGTM! Thanks!

And thanks for the inline comments. Those were really helpful.

**Pull request merged**